src/Servers/Kestrel/Kestrel/src/WebHostBuilderKestrelExtensions.cs

```diff
@@ -71,6 +112,26 @@ public static IWebHostBuilder UseKestrel(this IWebHostBuilder hostBuilder, Actio
         return hostBuilder.UseKestrel().ConfigureKestrel(options);
     }
 
+    /// <summary>
+    /// Specify Kestrel as the server to be used by the web host.
+    /// Includes less automatic functionality <see cref="UseKestrel(IWebHostBuilder, Action{KestrelServerOptions})"/> to make trimming more effective
+    /// (e.g. for Native AOT scenarios).  If the host ends up depending on some of the absent functionality, a best-effort
+    /// attempt will be made to enable it on-demand.  Failing that, an instructive error will be reported.
```

Comment:
I wonder how many people will stumble upon this doc comment when they're starting out and not have enough context to know what native AOT is or when the error will be reported (build time vs run time). May be worth linking to a native AOT doc? Or specify the extra detail in a <remarks> section.

Reply:
The exception happens during binding, but I'm assuming that's not a widely understood term (at least, by new users).  Would it be appropriate to describe it as "when `Build()` is invoked on the builder" (I think I have to be vague about which builder?)?

What doc did you have in mind for the link?  There's https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/, but (a) it's the top search result for "native aot" and (b) it says "It is currently not supported with ASP.NET Core". 😆

Reply:
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/native-aot?view=aspnetcore-8.0 seems like the right page, though it will need an aka.ms link.

Reply:
Updated the comment with the link and a comment about when the exception is thrown.